src/event_loop.rs

```diff
@@ -452,10 +453,23 @@ impl rwh_06::HasDisplayHandle for OwnedDisplayHandle {
     }
 }
 
+pub(crate) trait EventLoopProxyProvider: Send + Sync {
+    /// See [`EventLoopProxy::wake_up`] for details.
+    fn wake_up(&self);
+}
+
 /// Control the [`EventLoop`], possibly from a different thread, without referencing it directly.
 #[derive(Clone)]
 pub struct EventLoopProxy {
-    pub(crate) event_loop_proxy: platform_impl::EventLoopProxy,
+    pub(crate) proxy: Arc<dyn EventLoopProxyProvider>,
+}
+
+impl std::fmt::Debug for EventLoopProxy {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("EventLoopProxy")
+            .field("proxy", &(Arc::as_ptr(&self.proxy) as *const usize))
+            .finish_non_exhaustive()
```

Comment:
My issue is that I have no idea how a debug for object could be used. The pointer could at least show underlying identity, since just printing the name of the object sounds completely useless for any sort of debugging.

Reply:
> My issue is that I have no idea how a debug for object could be used.

```suggestion
        f.debug_struct("EventLoopProxy")
            .field("proxy", &self.proxy)
            .finish_non_exhaustive()
```
Seems to work just fine for me if `EventLoopProxyProvider` requires `Debug`.

> The pointer could at least show underlying identity, since just printing the name of the object sounds completely useless for any sort of debugging.

What is the "underlying identity"? How is that useful?

Its still useful to just print the name, it helps users just print whatever it is they need to print, e.g. if `EventLoopProxy` is part of a user struct, they don't have to add a custom `Debug` implementation. There is a reason why [the Rust guidelines specify that `Debug` should be implemented on every public type](https://rust-lang.github.io/api-guidelines/debuggability.html?highlight=debug#all-public-types-implement-debug-c-debug).

Reply:
>What is the "underlying identity"? How is that useful?

It could help in some cases, when backend returns unique pointer for each proxy, but I guess it's really easy to debug just by looking into source, since you have a single point to obtain the proxy.

Reply:
I'm not convinced that this is truly what most users intend to use `Debug` for. I am very strongly in favor of never outputting pointer addresses in `Debug`, unless the type is actually a pointer address.

In this case, I'm still in favor of just printing the name without actually requiring `Debug` from `EventLoopProxyProvider`, unless there is actually something useful to the user we want to show there.